config/pmd.xml

```diff
@@ -7,320 +7,313 @@
-    <!-- we use CheckstyleCustomShortVariable, to control length (will be fixed in PMD 5.4) and skip Override methods -->
-    <exclude name="ShortVariable"/>
-  </rule>
-<rule name="CheckstyleCustomShortVariable"
-      message="Avoid variables with short names that shorter than 2 symbols: {0}"
-      language="java"
-      class="net.sourceforge.pmd.lang.rule.XPathRule"
-      externalInfoUrl="">
-<description>
-   Fields, local variables, or parameter names that are very short are not helpful to the reader.
-</description>
-<priority>3</priority>
-<properties>
-   <property name="xpath">
-       <value>
-           <![CDATA[
- //VariableDeclaratorId[string-length(@Image) < 2]
- [not(ancestor::ForInit)]
- [not(../../VariableDeclarator and ../../../LocalVariableDeclaration and ../../../../ForStatement)]
- [not((ancestor::FormalParameter) and (ancestor::TryStatement))]
- [not(ancestor::ClassOrInterfaceDeclaration[//MarkerAnnotation/Name[pmd-java:typeof(@Image, 'java.lang.Override', 'Override')]])]
-            ]]>
-       </value>
-   </property>
-</properties>
-</rule>
-  <rule ref="rulesets/java/naming.xml/AbstractNaming">
+  <rule ref="category/java/errorprone.xml/DoNotCallSystemExit">
     <properties>
-     <!-- We can not brake compatibility with previous versions -->
-     <property name="violationSuppressXPath" value="//ClassOrInterfaceDeclaration[@Image='AbstractClassNameCheck' or @Image='AutomaticBean']"/>
+      <!-- It is ok to use System.exit in CLI class to report the exit code -->
+      <property name="violationSuppressXPath" value="//ClassOrInterfaceDeclaration[@Image='Main']"/>
     </properties>
   </rule>
-  <rule ref="rulesets/java/naming.xml/LongVariable">
-    <properties>
-      <!-- nothing bad in long and descriptive variable names if only they fit line, but still keep it reasonable -->
-      <property name="minimum" value="45"/>
-    </properties>
+
+  <rule ref="category/java/multithreading.xml">
+    <!-- Some of Checkstyle checks are not thread safe -->
+    <exclude name="UseConcurrentHashMap"/>
   </rule>
-  <rule ref="rulesets/java/naming.xml/ShortClassName">
+  <rule ref="category/java/multithreading.xml/AvoidSynchronizedAtMethodLevel">
     <properties>
-      <!-- Main is good name for class containing main method, Tag as inner class name is also fine -->
-      <property name="violationSuppressXPath" value="//ClassOrInterfaceDeclaration[@Image='Main' or @Image='Tag']"/>
+      <!-- UniqueProperties#put overloads synchronized method, so it should have synchronized
+           modifier -->
+      <property name="violationSuppressXPath" value="//ClassOrInterfaceDeclaration[@Image='UniqueProperties']//MethodDeclarator[@Image='put']"/>
     </properties>
   </rule>
 
-  <rule ref="rulesets/java/optimizations.xml">
-    <!--produces more false-positives than real problems-->
+  <rule ref="category/java/performance.xml">
+    <!-- Till https://github.com/checkstyle/checkstyle/issues/5638 -->
```

Comment:
is it the only rule that need extra suppression after upgrade to pmd6 ?
sound suspicious

Reply:
There was 5 of them.
Three in the `bestpractices` category, all of them were fixed with the issue #5633 
One about array::clone() for `com.puppycrawl.tools.checkstyle..api.Comment` class;
One about the loop in `com.puppycrawl.tools.checkstyle.JavadocDetailNodeParser`
One about using Test(expected = ...)

Two in the `performace` category:
One about FileInput/OutputStream classes: #5638 
One about toArray(new String[0]) 

There may be some other new rules. But they are not known since they did not report any violation.